### PR TITLE
Minor cleanup in some Facet tests

### PR DESCRIPTION
**Pull request by slow-J**
Fixing some IDE "warnings", such as removing some dead code + fixing a link + adding use of try-with-resources.

Did not fix `'search(org.apache.lucene.search.Query, org.apache.lucene.search.Collector)' is deprecated`, since it will be fixed as part of Issue: 12892.

While I am already touching these, let me know if anyone has strong opinions and I can also fix the `Statement lambda can be replaced with expression lambda` or `Redundant array creation for calling varargs method` "warnings".

### CODE REVIEW

**Comment by mikemccand** [comment]
> While I am already touching these, let me know if anyone has strong opinions and I can also fix the `Statement lambda can be replaced with expression lambda` or `Redundant array creation for calling varargs method` "warnings".

+1 to fix these too!

File: lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetValueSource.java [hunk 1]
```diff
@@ -694,7 +694,7 @@ private void checkResults(
                 "dim" + i,
                 new String[0],
                 aggregatedValue,
-                labelValues.toArray(new LabelAndValue[labelValues.size()]),
+                labelValues.toArray(new LabelAndValue[0]),
```
Comment (mikemccand) [inline lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetValueSource.java:697]: Hmm why this change?  Doesn't it mean we allocate both empty array and correctly sized array with this change, but before we were only allocating the one (correctly sized) array?

**Review by mikemccand — APPROVED** [review]
I just left one small comment about a `List.toArray` call.

File: lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetValueSource.java (hunk 1 above)
Comment (slow-J) [reply lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetValueSource.java:697]: We use the `labelValues.toArray(new LabelAndValue[0])` version in other tests as well as source code. I read a little online and this blog post is recommending the `labelValues.toArray(new LabelAndValue[0])` version due to optimizations in the JVM and JIT, making it more performant than the previous version: https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_new_reflective_array

**Comment by slow-J** [comment]
> > While I am already touching these, let me know if anyone has strong opinions and I can also fix the `Statement lambda can be replaced with expression lambda` or `Redundant array creation for calling varargs method` "warnings".
> 
> +1 to fix these too!

Thanks for the review Mike! I'll do these changes while I am at it.

**Comment by slow-J** [comment]
Hi @mikemccand can we merge or are we waiting for more reviewers.

**Review by stefanvodita — APPROVED** [review]

**Comment by stefanvodita** [comment]
I went ahead and merged since this PR had been pending for a few weeks. Thank you @slow-J for your contribution and @mikemccand for reviewing!

**Comment by mikemccand** [comment]
Thank you @slow-J and @stefanvodita!
